yaml.go

```diff
@@ -33,8 +33,19 @@ type JSONOpt func(*json.Decoder) *json.Decoder
 // Unmarshal converts YAML to JSON then uses JSON to unmarshal into an object,
 // optionally configuring the behavior of the JSON unmarshal.
 func Unmarshal(y []byte, o interface{}, opts ...JSONOpt) error {
+	return unmarshal(yaml.Unmarshal, y, o, opts)
+}
+
+// UnmarshalStrict is like Unmarshal except that any mapping keys that are
+// duplicates, will result in an error.
+// It combines well with the option DisallowUnknownFields.
```

Comment:
This is a somewhat odd and subjective statement.

Let's just say:

    // To also be strict about unknown fields, use the DisallowUnknownFields option.

(but that makes me wonder why duplicate strictness isn't just an option)

Reply:
> but that makes me wonder why duplicate strictness isn't just an option

The signature of the existing unmarshalling function is:
`func Unmarshal(y []byte, o interface{}, opts ...JSONOpt) error`
I designed my new function to be consistent with the existing one:
`func UnmarshalStrict(y []byte, o interface{}, opts ...JSONOpt) error`.

In addition, passing `JSONOpt`s as argument helps to keep this code backward-compatible with older Go versions since `DisallowUnknownFields` is only available in newer versions.